2015-Fall/soerendalby/Calculater/ViewController.swift

```diff
@@ -26,5 +27,53 @@ class ViewController: UIViewController {
         
     }
     
+    @IBAction func operate(sender: UIButton) {
+        let operation = sender.currentTitle!;
+        
+        if continueCurrentNumber {
+            enter();
+        }
+        
+        var result:Double?;
+        
+        switch operation {
+    
+        case "x":
+            result = calculaterBrain.performOperation(CalculatorBrain.Operation.Multiply)
+        case "/":
+            result = calculaterBrain.performOperation(CalculatorBrain.Operation.Divide)
+        case "+":
+            result = calculaterBrain.performOperation(CalculatorBrain.Operation.Plus)
+        case "-":
+            result = calculaterBrain.performOperation(CalculatorBrain.Operation.Minus)
+        default:
+            // should cause fatal error
+            break;
+        }
+        if result != nil {
```

Comment:
This could be replaced by the ``if let`` construct (if you want to keep the ``result`` variable)